File: My_MVVM_Login_Module_Java/src/main/java/com/mobitant/my_mvvm_login_module_java/MainActivity.java
package com.mobitant.my_mvvm_login_module_java;

import android.arch.lifecycle.ViewModelProviders;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.Toast;
import com.mobitant.my_mvvm_login_module_java.Interface.LoginResultCallbacks;
import com.mobitant.my_mvvm_login_module_java.databinding.ActivityMainBinding;
import es.dmoral.toasty.Toasty;

public class MainActivity extends AppCompatActivity implements LoginResultCallbacks {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMainBinding activityMainBinding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        activityMainBinding.setViewModel(ViewModelProviders.of(
                this,
                new LoginViewModelFactory(this))
                .get(LoginViewModel.class
        ));
    }

    @Override
    public void onSuccess(String message) {
        Toasty.success(this,"Login success", Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onError(String message) {
        Toasty.error(this,"Login error",Toast.LENGTH_SHORT).show();
    }
}

File: My_MVP_Login_Module/src/main/java/com/mobitant/android_architecture/View/MainActivity.kt
package com.mobitant.android_architecture.View

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Toast
import com.mobitant.android_architecture.Presenter.LoginPresenter
import com.mobitant.android_architecture.Presenter.MyLoginPresenter
import com.mobitant.android_architecture.R
import es.dmoral.toasty.Toasty
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(),MyLoginView {
    override fun onLoginSuccess(message: String) {
        Toasty.success(this,message,Toast.LENGTH_SHORT).show()
    }
    override fun onLoginError(message: String) {
        Toasty.error(this,message,Toast.LENGTH_SHORT).show()
         }

    internal lateinit var loginPresenter: MyLoginPresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        loginPresenter = LoginPresenter(this)

        btn_login.setOnClickListener {
            loginPresenter.onLogin(edt_email.text.toString(),edt_password.text.toString())
        }
    }
}
